### ISSUE TITLE
no-duplicate-variable allows duplicate let declarations in the same block scope

### ISSUE THREAD

**Issue by wereHamster** [palantir/tslint#321]
The following is accepted even though it should generate a warning.

```javascript
function test() {
    let x = 1;
    let x = 2;
}
``` 

Related issues: #295, #303.

**Comment by adidahiya** [comment]
Compiling that particular example via `tsc` throws an error. We hope to avoid duplicating the work of the compiler in TSLint, so I don't think we're going to directly target that example.
